src/interfaces/simulation.ts

```diff
@@ -38,84 +44,145 @@ export class SimulationInterface<T extends ChainId> extends ServiceInterface<T>
     options: SimulationOptions = {}
   ): Promise<TransactionOutcome> {
     const signer = this.ctx.provider.write.getSigner(from);
-    const zapProtocol = PickleJars.includes(toVault) ? ZapProtocol.PICKLE : ZapProtocol.YEARN;
-    const vaultContract =
-      zapProtocol === ZapProtocol.PICKLE
-        ? new PickleJarContract(toVault, signer)
-        : new YearnVaultContract(toVault, signer);
-
-    const underlyingToken = await vaultContract.token().catch(() => {
-      throw new EthersError("failed to fetch token", EthersError.FAIL_TOKEN_FETCH);
-    });
-    const isZapping = underlyingToken !== sellToken;
-    let forkId: string | undefined;
-    let simulateDeposit: (save: boolean) => Promise<TransactionOutcome>;
 
-    if (isZapping) {
-      if (!options.slippage) {
-        throw new SdkError("slippage needs to be specified for a zap", SdkError.NO_SLIPPAGE);
-      }
+    const vaultContract = this.getVaultContract({ toVault, signer });
 
-      let needsApproving: boolean;
+    const vaultMetadata = await this.yearn.services.meta.vault(toVault);
 
-      if (sellToken === EthAddress) {
-        needsApproving = false;
-      } else {
-        needsApproving = await this.yearn.services.zapper
-          .zapInApprovalState(from, sellToken, zapProtocol)
-          .then((state) => !state.isApproved)
-          .catch(() => {
-            throw new ZapperError("approval state", ZapperError.ZAP_IN_APPROVAL_STATE);
+    // TODO Fetch the token
+    const token = {};
+
+    const { isZapInSupported, zapInWith } = getZapInOptions({ chainId: this.chainId, token, vaultMetadata });
+
+    if (isZapInSupported) {
+      switch (zapInWith) {
+        case "zapperZapIn": {
```

Comment:
should we have this as constants?

Reply:
They are constants

<img width="1277" alt="Screen Shot 2022-04-20 at 9 19 23 am" src="https://user-images.githubusercontent.com/78794805/164116645-b8f40ad5-b038-41e4-8917-7ad4c0b4cd6a.png">

Reply:
Right but I mean define a variable with this constans, so we don't use the strings directly in case they change on the furure. so like we have at some part of the code `const ZAPPER_ZAP_IN = "zapperZapIn"` and we just use ZAPPER_ZAP_IN here

Reply:
If the string changes then lint would warn us, assigning to const doesn't really help here. The alternative would be enums, but Gambito prefers this way.

Reply:
Yeah, agree with Chef here that assigning to a `const` is not needed since ts compiler would indicate when a non valid string is used. And yes, we are avoiding Enums for convenience.